src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+				}
+				break;
+			case IF_ICMPNE:
+				if ((int) methodContext.pop() != (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPLT:
+				if ((int) methodContext.pop() > (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPGE:
+				if ((int) methodContext.pop() <= (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPGT:
+				if ((int) methodContext.pop() < (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ICMPLE:
+				if ((int) methodContext.pop() >= (int) methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ACMPEQ:
+				if (methodContext.pop() == methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case IF_ACMPNE:
+				if (methodContext.pop() != methodContext.pop()) {
+					branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				}
+				break;
+			case GOTO:
+				branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				break;
+			case JSR:
+				branchOffset = ByteUtils.readShort(instructions, currentInstruction + 1);
+				methodContext.push(currentInstruction + 3);
+				break;
+			case JSR_W:
+				branchOffset = ByteUtils.readInt(instructions, currentInstruction + 1);
+				methodContext.push(currentInstruction + 5);
+				break;
+			case RET:
+				index = ByteUtils.readUnsignedByte(instructions, currentInstruction + 1);
+				currentInstruction = (int) methodContext.load(index);
+				break;
+			case TABLESWITCH:
+				//todo
+				break;
+			case LOOKUPSWITCH:
+				//todo
```

Should probably throw an UnsupportedOperationException here